com.sap.cloud.lm.sl.persistence/src/main/java/com/sap/cloud/lm/sl/persistence/services/ProgressMessageService.java

```diff
@@ -170,6 +171,38 @@ public Integer execute(Connection connection) throws SQLException {
         }
     }
 
+    public int removeAllByProcessIds(final List<String> processIds) throws SLException {
+        SqlExecutor<Integer> executor = new ProgressMessageSqlExecutor<Integer>();
+        try {
+            return executor.execute(new StatementExecutor<Integer>() {
+                @Override
+                public Integer execute(Connection connection) throws SQLException {
+                    int rowsRemoved = 0;
+                    PreparedStatement statement = null;
+                    try {
+                        connection.setAutoCommit(false);
+                        statement = connection.prepareStatement(getQuery(DELETE_MESSAGE_BY_PROCESS_ID, tableName));
+                        for (String processId : processIds) {
+                            statement.setString(1, processId);
+                            statement.addBatch();
+                        }
+                        int[] rowsRemovedArray = statement.executeBatch();
+                        rowsRemoved = Arrays.stream(rowsRemovedArray).sum();

```

Comment:
Again - use only "rowsRemovedArray.length".

Reply:
Same as above.

Reply:
As previous - we will leave it as it is.